awsutil/generate_credentials.go

```diff
@@ -230,6 +234,22 @@ func (c *CredentialsConfig) GenerateCredentialChain(opt ...Option) (*credentials
 				// Add the web identity role credential provider
 				providers = append(providers, webIdentityProvider)
 			}
+		} else if c.WebIdentityToken != "" {
```

Comment:
The `WebIdentityTokenFile` option has functional options, a small bit of validation, and an environment variable associated with it. It feels like we should probably have the first two for `WebIdentityToken` too for consistency, and maybe the third? Although I'm slightly inclined to omit the environment variable given the expected consumption pattern as far as I understand it.

Reply:
Yep, thanks for pointing that out. Updated with the option and validation in 02a10db9112506b25905a3a4eda8ce0d776eb9b4 and 3bde4d9